src/likelihoods/categorical.jl

```diff
@@ -0,0 +1,16 @@
+"""
+    CategoricalLikelihood
+
+Categorical likelihood is to be used if we assume that the 
+uncertainity associated with the data follows a Categorical distribution.
+```math
+    p(y|f_1, f_2, \\dots, f_n) = Categorical(y | softmax(f_1, f_2, \\dots, f_n))
```

Comment:
I just noticed that this will create unidentifiability issues (the resulting distribution is invariant under additions of constant vectors). These could be resolved by e.g. defining instead
```suggestion
    p(y | f_2, f_3, \\dots, f_n) = Categorical(y | softmax(1, f_2, \\dots, f_n))
```
(basically `f_1` is fixed to 1).

Reply:
Isn't this a bit non-intuitive? Is there no other way to avoid in-variances?

Reply:
That's the common approach, e.g., in [multinomial logistic regression](https://en.wikipedia.org/wiki/Multinomial_logistic_regression) (on Wikipedia `f_n` is set to 1 instead of `f_1`).

Reply:
It's in the same way that for classifying a binary case you only need one GP, not two.
I would however be in favor of keeping the convention for the wiki page : 
```suggestion
    p(y|f_1, f_2, \\dots, f_n) = Categorical(y | softmax(f_1, f_2, \\dots, f_{n-1}, 1))
```

Reply:
I was just checking other packages to see what they have done.

GPFlow doesn't seem to have an equivalent implementation for Categorical likelihood. In GPyTorch, I think `SoftmaxLikelihood` is the equivalent. https://docs.gpytorch.ai/en/latest/_modules/gpytorch/likelihoods/softmax_likelihood.html#SoftmaxLikelihood

Weirdly they don't seem to be worried about invariances but they optionally support "mixing weights".

Reply:
My question was: Do we want to support something like to "mixing weights"?

Reply:
It's a bit weird from this line 
`mixed_fs = function_samples @ self.mixing_weights.t()`
It sounds like they just make a linear combinations of the output (already what we can do with the kernel basically), I would not consider it.

Reply:
BTW it should of course be `0` in the softmax function (I'm not sure why I wrote 1):
```suggestion
    p(y|f_1, f_2, \\dots, f_{n-1}) = Categorical(y | softmax(f_1, f_2, \\dots, f_{n-1}, 0))
```